add: LevelSelection Editor Window

realTobby opened the pull request:
implemented basic level selection window according to https://manuel-rauber.com/2022/03/14/unity-tip-level-selection-editor-window/ - can be found under "Community Project" => "Level Selection"

**Beschreibung**
Bare Bones Level Selection EditorWindow Implementiert, Szenen können per Button Klick einzeln oder addative geöffnet werden. Siehe => #33 

## Checkliste
<!-- Bitte lösche diese Liste nicht. Du kannst sie nach dem Stellen des PRs abhaken. -->

- [x] Aufgabe verlinkt (falls nicht, PR editieren)
- [ ] Tests geschrieben (nur relevant für Code-Änderungen)
- [ ] Dokumentation aktualisiert (nur relevant für System-Entwicklungen)

## PR Typ
<!-- Bitte lösche diese Liste nicht. Du kannst sie nach dem Stellen des PRs abhaken. -->

- [ ] Bugfix
- [x] Feature
- [ ] 3D-Modell
- [ ] 2D-Arbeit
- [ ] Sound-Effekte
- [ ] Musik
- [ ] Dokumentation
- [ ] Sonstiges, bitte beschreiben:

realTobby committed:
add: LevelSelection Editor Window

implemented basic level selection window according to https://manuel-rauber.com/2022/03/14/unity-tip-level-selection-editor-window/ - can be found under "Community Project" => "Level Selection"

realTobby committed:
fix: show variable Manager Submenu

showManagers statt showMenus for Managers submenu

ManuelRauber commented on CommunityProject/Assets/_Game/Scripts/Runtime/Constants.cs:
```diff
@@ -8,6 +8,7 @@ public static class MenuNames
 			public const string Events = MenuName + "/Events";
 			public const string SceneManagement = MenuName + "/Scene Management";
 			public const string UI = MenuName + "/UI";
+			public const string LevelSelection = MenuName + "/Level Selection";
```
Es reicht, wenn wir diese Konstante nur in den Editor-Konstanten definieren, da wir's zur Laufzeit eigentlich hier nicht brauchen. :)

realTobby commented:
@ManuelRauber Wäre es vielleicht eine gute Idee, die Menüs dynamisch zu erstellen? Dann müssten wir das EditorWindow nicht mehr anpassen später. Knotenpunkt ist der Ordner innerhalb Scenes, die Buttons sind die Scenes im jeweiligen Ordner.
Ist das verständlich? :D (Wir müssen also nur dafür sorgen das die Ordner Struktur immer stimmt)

ManuelRauber commented:
> @ManuelRauber Wäre es vielleicht eine gute Idee, die Menüs dynamisch zu erstellen? Dann müssten wir das EditorWindow nicht mehr anpassen später. Knotenpunkt ist der Ordner innerhalb Scenes, die Buttons sind die Scenes im jeweiligen Ordner.
> Ist das verständlich? :D (Wir müssen also nur dafür sorgen das die Ordner Struktur immer stimmt)

Das mit den Menüs machen wir, sobald wir erste Levels/Packs haben. 
Mache ich bei [Bug-A-Ball](https://bugaball.com) auch so, ein paar sind statisch drin und alle Level etc. kommen dynamisch hinzu.

Ich hab beim PrefabManager das ganze mal mit Addressables ausprobiert, das scheint zu funktionieren. Eventuell bauen wir das LevelWindow dann auch auf Addressables um, dann ist's weniger fehleranfällig durch die Ordnerstruktur.

ManuelRauber commented on CommunityProject/Assets/_Game/Scripts/Editor/Menus/LevelSelectionMenu/LevelSelectionEditorWindow.cs:
```diff
@@ -0,0 +1,144 @@
+{
+    public class LevelSelectionEditorWindow : EditorWindow
+    {
+		#region Private Fields
```
Bitte keinerlei `#region` nutzen, siehe [Coding Conventions](https://github.com/BoundfoxStudios/community-project/blob/develop/docs/contributing/technical/coding-conventions.md#kommentare)

ManuelRauber commented on CommunityProject/Assets/_Game/Scripts/Editor/Menus/LevelSelectionMenu/LevelSelectionEditorWindow.cs:
```diff
@@ -0,0 +1,144 @@
+    public class LevelSelectionEditorWindow : EditorWindow
+    {
+		#region Private Fields
+		private bool _openAdditive;
```
Hier passt irgendwie die Einrückung nicht.

ManuelRauber commented on CommunityProject/Assets/_Game/Scripts/Editor/Menus/LevelSelectionMenu/LevelSelectionEditorWindow.cs:
```diff
@@ -0,0 +1,144 @@
+using BoundfoxStudios.CommunityProject.SceneManagement;
```
Diese Datei liegt generell im falschen Ordner.

Aktuell liegt sie unter `Editor/Menus/LevelSelectionMenu`.

Da es aber ein Window ist, sollte es eher unter `Editor/Windows` liegen. Eigenen weiteren Ordner braucht's dort drin dann nicht, besteht ja aktuell aus einer Datei.

ManuelRauber commented on CommunityProject/Assets/_Game/Scripts/Editor/Menus/LevelSelectionMenu/LevelSelectionEditorWindow.cs:
```diff
@@ -0,0 +1,144 @@
+			EndFoldoutGroup("Managers", showManagers);
+		}
+
+		private void RenderGameplay()
```
Kann aktuell noch entfernt werden, macht ja nichts sinnvolles.

ManuelRauber commented on CommunityProject/Assets/_Game/Scripts/Editor/Menus/LevelSelectionMenu/LevelSelectionEditorWindow.cs:
```diff
@@ -0,0 +1,144 @@
+				OpenSceneButton("PersistentManagers", "Managers/PersistentManagers");
+				EditorGUILayout.EndHorizontal();
+			}
+			EndFoldoutGroup("Managers", showManagers);
```
Den Part könnte man in eine eigene Methode auslagern und dann bei `RenderScenes` aufrufen.

Dann könnte man jeweils auch die `show*`-Variablen wieder auf ein `show` umbenennen.

ManuelRauber reviewed (changes requested):
Ein paar Kleinigkeiten zur Korrektur. :)

realTobby committed:
fix: Issue Comments

Moved LevelSelectionEditorWindow-FIle to Editor/Windows, moved constant to editor constant, removed regions, removed RenderGamePlay-Function, removed unnesseary-usings, seperate RenderScenes into sub-functions

realTobby committed:
fix: formatting

formatting

ManuelRauber commented on CommunityProject/Assets/_Game/Scripts/Editor/Windows/LevelSelectionEditorWindow.cs:
```diff
@@ -0,0 +1,117 @@
+using UnityEditor.SceneManagement;
+using UnityEngine;
+
+namespace BoundfoxStudios.CommunityProject.Editor.Windows.LevelSelectionEditorWindow
```
```suggestion
namespace BoundfoxStudios.CommunityProject.Editor.Windows
```

Namespaces sind nur auf Ordner-Ebene, d.h. es endet hier beim Ordner "Windows" :)

ManuelRauber reviewed (changes requested):
Eine letzte kleine Änderung :)

realTobby committed:
fix: namespace

removed classname inside of namespace

realTobby committed:
fix: newline between function

added a newline between functions

ManuelRauber commented:
Herzlichen Dank! :)

ManuelRauber commented:
@all-contributors please add @realTobby for code

ManuelRauber commented:
@all-contributors please add @realTobby for code
